Java/src/main/java/net/hypixel/api/reply/PlayerReply.java

```diff
@@ -1,23 +1,358 @@
 package net.hypixel.api.reply;
 
+import com.google.gson.JsonArray;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
+import java.util.Date;
+import java.util.UUID;
+import net.hypixel.api.util.ILeveling;
 
 public class PlayerReply extends AbstractReply {
-    private JsonElement player;
 
-    public JsonObject getPlayer() {
-        if (player == null || player.isJsonNull()) {
-            return null;
-        } else {
-            return player.getAsJsonObject();
-        }
+    private static final Player BLANK_PLAYER = new Player();
+
+    private Player player;
+
+    public Player getPlayer() {
+        return player == null ? BLANK_PLAYER : player;
     }
 
     @Override
     public String toString() {
         return "PlayerReply{" +
-                "player=" + player +
-                "} " + super.toString();
+            "player=" + player +
+            "} " + super.toString();
+    }
+
+    public static class Player {
+
+        private static final String DEFAULT_RANK = "NONE";
+
+        private JsonElement raw;
+
+        /**
+         * @return A string representing the player's Minecraft UUID (without hyphens)
+         */
+        public String getUuid() {
+            return getStringProperty("uuid", null);
+        }
+
+        /**
+         * @return The Minecraft username the player had when they last connected to Hypixel
+         */
+        public String getName() {
+            return getStringProperty("displayname", null);
+        }
+
+        /**
+         * @return The total amount of network experience earned by this player
+         */
+        public long getNetworkExp() {
+            return getNumberProperty("networkExp", 0).longValue();
```

Comment:
This should take `networkLevel` into account
```py
    @property
    def exp(self):
        exp = self.get('networkExp', 0)
        exp += leveling.get_total_exp_to_full_level(self.get('networkLevel', 0) + 1)
        return exp
```

Reply:
Didn't know about that, thanks.

Reply:
Yes, any level calculations should use this exp.